src/hooks.js

```diff
@@ -0,0 +1,32 @@
+// @flow
+
+import type {Config} from 'cli-engine-config'
+import path from 'path'
+
+export default class Hooks {
+  config: Config
+  hooks: {[name: string]: string[]}
+
+  constructor ({config}: {config: Config}) {
+    this.config = config
+    this.hooks = {}
+    let cliEngine: any = config.pjson['cli-engine'] || {}
+    for (let [name, script] of Object.entries(cliEngine['hooks'] || {})) {
+      if (typeof script !== 'string') throw new Error(`expected ${name} to be a string`)
+      this.register(name, script)
+    }
+  }
+
+  register (name: string, script: string) {
+    this.hooks[name] = this.hooks.name || []
+    this.hooks[name].push(script)
+  }
+
+  async run (name: string, args: Object) {
+    for (let script of this.hooks[name] || []) {
+      let hook = require(path.join(this.config.root, script))
+      args.config = this.config
+      await hook(args)
```

Comment:
I thought I had a good example of why we might not want to terminate the CLI on a single hook failure, but now I'm not really coming up with one, aside from bad design/implementation in the hook code.  Probably not worth adding the ability to configure the behavior until we have something concrete to work with.

I believe what I was originally imaging was something where multiple plugins' hooks could all run in one CLI invocation -- in a case like that you wouldn't want a poorly implemented third party plugin to interrupt the function of an official plugin, but now I'm not sure such hooks would ever exist.